### PR TITLE
4724 new toast design

### PR DESCRIPTION
**Pull request by andrewkryshtal**

### CODE REVIEW

**Review by keshan3262 — APPROVED** [review]

File: src/styles/colors.ts [hunk 1]
```diff
@@ -51,18 +51,27 @@ const backgroundLightColors = {
   cardBG: '#ffffff',
   lines: '#e4e4e4',
   disabled: '#cbcbcb',
-  input: '#f0f0f0'
+  input: '#f0f0f0',
+  toastBG: '#ffffff',
```
Comment (unixvb) [inline src/styles/colors.ts:55]: reuse `white` color

File: src/styles/colors.ts [hunk 2]
```diff
@@ -51,18 +51,27 @@ const backgroundLightColors = {
...
+  successToastOverlay: basicLightColors.adding,
+  warningToastOverlay: hexa('#FF7A00', 0.1),
+  errorToastOverlay: basicLightColors.destructive
```
Comment (unixvb) [inline src/styles/colors.ts:58]: create new `peach` color and reuse `adding` & `destructive`

File: src/styles/colors.ts [hunk 3]
```diff
@@ -51,18 +51,27 @@ const backgroundLightColors = {
...
-  input: '#2D2D2D'
+  input: '#2D2D2D',
+  toastBG: '#000000',
+  successToastOverlay: basicLightColors.adding,
+  warningToastOverlay: hexa(basicDarkColors.black, 0.1),
+  errorToastOverlay: basicDarkColors.destructive
 };
 
-const lightTheme = {
+const lightTheme: Record<string, string> = {
```
Comment (unixvb) [inline src/styles/colors.ts:74]: remove type

File: src/toast/toast/custom-toast.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import { formatSize } from '../../styles/format-size';
+import { useColors } from '../../styles/use-colors';
+import { conditionalStyle } from '../../utils/conditional-style';
+import { isDefined } from '../../utils/is-defined';
+import { isString } from '../../utils/is-string';
+import { tzktUrl } from '../../utils/linking.util';
+import { useToastStyles } from './toast.styles';
+
+interface Props {
+  onPress?: () => void;
```
Comment (unixvb) [inline src/toast/toast/custom-toast.tsx:20]: props order, `EmptyFn`, default value

File: src/toast/toast/custom-toast.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  props?: {
+    operationHash?: string;
+  };
+}
+
+export const CustomToast: FC<Props> = ({ onPress, text1, text2, hide, toastType, props }) => {
+  const styles = useToastStyles(toastType)();
+  const colors = useColors();
+
+  const onPressHandler = () => {
```
Comment (unixvb) [inline src/toast/toast/custom-toast.tsx:34]: `handlePress`

File: src/toast/toast/custom-toast.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    }
+    hide();
+  };
+
+  return (
+    <TouchableOpacity style={styles.container} onPress={onPressHandler}>
+      <View style={styles.overlay}>
+        <View style={styles.innerContent}>
+          <Icon
+            name={isString(text1) ? IconNameEnum.Success : IconNameEnum.AlertShield}
```
Comment (unixvb) [inline src/toast/toast/custom-toast.tsx:46]: condition should be related to toast type

File: src/toast/toast/custom-toast.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+            color={toastType === ToastTypeEnum.Warning ? undefined : colors.white}
```
Comment (unixvb) [inline src/toast/toast/custom-toast.tsx:47]: ...(toastType !== ToastTypeEnum.Warning && {color: colors.white})

File: src/toast/toast/toast.styles.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      backgroundColor: colors.toastBG
+    },
+    overlay: {
+      borderRadius: formatSize(8),
+      backgroundColor: colors[`${toastType}ToastOverlay`],
+      flex: 1,
+      justifyContent: 'center'
+    },
+    innerContent: {
+      display: 'flex',
```
Comment (unixvb) [inline src/toast/toast/toast.styles.ts:22]: remove `flex`

File: src/styles/colors.ts [hunk 9]
```diff
@@ -51,15 +53,18 @@ const backgroundLightColors = {
   cardBG: '#ffffff',
   lines: '#e4e4e4',
   disabled: '#cbcbcb',
-  input: '#f0f0f0'
+  input: '#f0f0f0',
+  toastBG: basicLightColors.white
```
Comment (unixvb) [inline src/styles/colors.ts:57]: remove `toastBG` use `white` color instead

File: src/toast/toast-provider.tsx [hunk 10]
```diff
@@ -6,9 +6,37 @@ import { EmptyFn } from '../config/general';
...
+      hide={hide}
+      onPress={onPress}
+    />
+  ),
+  [ToastTypeEnum.Error]: ({ hide, text1, text2, onPress }: ToastProps) => (
+    <CustomToast text1={text1} text2={text2} onPress={onPress} hide={hide} toastType={ToastTypeEnum.Error} />
+  ),
+  [ToastTypeEnum.Warning]: ({ hide, text1, text2, onPress }: ToastProps) => (
+    <CustomToast text1={text1} text2={text2} onPress={onPress} hide={hide} toastType={ToastTypeEnum.Warning} />
+  )
```
Comment (unixvb) [inline src/toast/toast-provider.tsx:39]: `onPress` order

File: src/toast/toast-provider.tsx [hunk 11]
```diff
@@ -6,9 +6,37 @@ import { EmptyFn } from '../config/general';
 import { ToastTypeEnum } from '../enums/toast-type.enum';
 import { CopiedToast } from './copied-toast/copied-toast';
 import { ToastProviderStyles } from './toast-provider.styles';
+import { CustomToast } from './toast/custom-toast';
+
+interface ToastProps {
+  hide: EmptyFn;
+  text1: string;
+  text2: string;
+  onPress?: () => void;
```
Comment (unixvb) [inline src/toast/toast-provider.tsx:15]: use `EmptyFn`

File: src/toast/toast/custom-toast.tsx [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+import { hexa } from '../../utils/style.util';
+import { useToastStyles } from './toast.styles';
+
+interface Props {
+  onPress?: EmptyFn;
+  text1?: string;
+  text2: string;
+  hide: EmptyFn;
+  toastType: ToastTypeEnum;
+  operationHash?: string;
```
Comment (unixvb) [inline src/toast/toast/custom-toast.tsx:25]: props order

File: src/toast/toast/custom-toast.tsx [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+export const CustomToast: FC<Props> = ({ onPress, text1, text2, hide, toastType, operationHash }) => {
```
Comment (unixvb) [inline src/toast/toast/custom-toast.tsx:28]: props order

File: src/toast/toast/custom-toast.tsx [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+  const styles = useToastStyles();
+  const colors = useColors();
+
+  const backgroundColorMap: { [key: string]: string } = {
```
Comment (unixvb) [inline src/toast/toast/custom-toast.tsx:32]: use `Record`

File: src/toast/toast/custom-toast.tsx [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+import { useColors } from '../../styles/use-colors';
+import { isDefined } from '../../utils/is-defined';
+import { isString } from '../../utils/is-string';
+import { tzktUrl } from '../../utils/linking.util';
+import { hexa } from '../../utils/style.util';
+import { useToastStyles } from './toast.styles';
+
+interface Props {
+  onPress?: EmptyFn;
+  text1?: string;
```
Comment (unixvb) [inline src/toast/toast/custom-toast.tsx:21]: `title`

File: src/toast/toast/custom-toast.tsx [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+  text2: string;
```
Comment (unixvb) [inline src/toast/toast/custom-toast.tsx:22]: `description`

File: src/toast/toast-provider.tsx [hunk 17]
```diff
@@ -6,9 +6,37 @@ import { EmptyFn } from '../config/general';
 import { ToastTypeEnum } from '../enums/toast-type.enum';
 import { CopiedToast } from './copied-toast/copied-toast';
 import { ToastProviderStyles } from './toast-provider.styles';
+import { CustomToast } from './toast/custom-toast';
+
+interface ToastProps {
+  hide: EmptyFn;
+  text1: string;
```
Comment (unixvb) [inline src/toast/toast-provider.tsx:13]: `title`

File: src/toast/toast.utils.ts [hunk 18]
```diff
@@ -1,19 +1,40 @@
 import Toast from 'react-native-toast-message';
 
+import { EmptyFn } from '../config/general';
 import { ToastTypeEnum } from '../enums/toast-type.enum';
 
-export const showErrorToast = (title: string, description?: string) =>
+interface ToastProps {
+  text: string;
```
Comment (unixvb) [inline src/toast/toast.utils.ts:7]: `description`

File: src/toast/toast.utils.ts [hunk 19]
```diff
@@ -1,19 +1,40 @@
...
+  title?: string;
+  onPress?: EmptyFn;
+  operationHash?: string;
+}
+
+export const showErrorToast = ({ text = '', title, onPress }: ToastProps) =>
```
Comment (unixvb) [inline src/toast/toast.utils.ts:13]: no need for the initial value, `text` is required param

Comment (unixvb) [reply src/toast/toast.utils.ts:13]: the same in other places

**Review by unixvb — APPROVED** [review]

**Pull request merged**